src/main/scala/ayai/systems/ai/QuestGenerationSystem.scala

```diff
@@ -22,55 +24,73 @@ object QuestGenerationSystem {
-              significantMemory = memory
-            }
-          }
 
-          // next, we can decide whether that thing is "negative" or "positive". If it's negative, the NPC will want that
-          // thing to be killed or something. If it's positive, maybe you should bring it to them.
-          /*if ( significantMemory != null ) {
-            if ( significantMemory.relationship > 0 ) {
-              // positive relationship quests
-              objectives += new FetchObjective( "Thing To Get", "Bring it to this thing" )
-            }else{
-              // negative relationship quests.
-              objectives += new KillObjective( "Get Entity Name from Passed ID Here", 0, 10 )
-            }
-          }*/
-
-          // populate it with objectives
-          objectives += new KillObjective("ENEMY ID HERE",10,0)
-
-          // create a blank quest which we'll populate in a minute.
-          val questComponent = new Quest(
-            1,                    // id
-            "hello",              // title
-            "do a thing",         // description
-            12,                   // recommended level
-            objectives.toList     // objectives
-          )
-
-          // add a new quest to the quest bag of the initiator!
-          genQuest.initiator.getComponent(classOf[QuestBag]).asInstanceOf[QuestBag].addQuest( questComponent )
-          // the gamestate needs to keep a list of our quests?
-          //networkSystem.actorSelection("user/QuestMap") ! AddQuest("QUEST"+questData.id, questComponent)
-
-          // lastly, remove the generation request. We don't need it anymore.
-          e.removeComponent( classOf[GenerateQuest] )
+          // the next thing we'll need is the memory of the quest-giver. we'll use their memories to figure out what the
+          // quest should target.
+          genQuest.initiator.getComponent(classOf[Memory]) match {
+            case Some(initiatorMemory: Memory) =>
+
+              //val recipientHistory = genQuest.recipient.getComponent(classOf[QuestHistory])
+
+              // create an objective list
+              var objectives = new ListBuffer[Objective]()
+
+              // next, we need to populate our quest with objectives. To do this, we'll find the most "significant" thing
+              // in an NPC's memory and build objectives based on that.
+              var mostSignificantMemory: MemoryContents = null
+              for (memory <- initiatorMemory.entitiesRemembered) {
+                if (math.abs(memory.relationship) > math.abs(mostSignificantMemory.relationship)) {
+                  mostSignificantMemory = memory
+                }
+              }
+
+              // next, we can decide whether that thing is "negative" or "positive". If it's negative, the NPC will want that
+              // thing to be killed or something. If it's positive, maybe you should bring it to them.
+              if ( mostSignificantMemory != null ) {
```

Comment:
if you take the above refactoring, `mostSignificantMemory` is an `Option[MemoryContents]` which you can just `map` over to do the stuff you need to do if it's non-empty.